.github/workflows/l10n_extract.yaml

```diff
@@ -0,0 +1,64 @@
+name: Extract Strings
+on:
+  schedule:
+    - cron: '0 22 * * 4' # At 15:00 PDT on Thursday
+  workflow_dispatch:
+jobs:
+  extract:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Linux packages
+        run: |
+          sudo apt update
+          sudo apt install gettext graphicsmagick git hub translate-toolkit -y
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14
+      - name: Install global npm packages
+        run: |
+          npm install -g grunt-cli
+      - name: Clone l10n repository
+        uses: actions/checkout@v3
+        with:
+          path: "fxa-l10n"
+      - name: Clone FxA code repository
+        uses: actions/checkout@v3
+        with:
+          repository: "mozilla/fxa"
+          fetch-depth: 1
+          path: "fxa-code"
+      - name: Install npm packages
+        run: |
+          cd fxa-l10n
+          npm install
+      - name: Extract strings
+        run: |
+          cd fxa-code
+          yarn workspaces focus fxa-content-server fxa-auth-server fxa-payments-server fxa-settings
+          yarn workspace fxa-settings build
+          yarn workspace fxa-auth-server grunt merge-ftl
+
+          NODE_ENV=development ../fxa-l10n/scripts/extract_strings.sh \
+            --mailer-repo ./packages/fxa-auth-server \
+            --payments-repo ./packages/fxa-payments-server \
+            --content-repo ./packages/fxa-content-server \
+            --settings-repo ./packages/fxa-settings \
+            --l10n-repo ../fxa-l10n
+      - name: Commit changes and open pull request
+        run: |
+          git config --global user.email 'bugmirror@restmail.net'
+          git config --global user.name 'Bug Mirror'
+          # Random release number to avoid collision with old trains or branches
+          TRAIN_NUMBER=$(( $RANDOM + $RANDOM + 1000 ))
+
+          cd fxa-l10n
+          git checkout -B "merge-train-$TRAIN_NUMBER-strings"
```

Comment:
I moved all the git commands out of the bash script, so no need to pass the random train number as parameter.